File: DeliveryIP_GitHub/dstoolkit/app/src/types/blogEntry.ts
import { BlogEntryStatus } from "./blogEntryStatus";
import { BlogEntryType } from "./blogEntryType";

export interface BlogEntry {
    id: string;
    title: string;
    heroImage: {
        path: string;
        alternativeText: string;
        base64?: string;
    };
    introduction: string;
    createdBy: {
        on: Date;
        displayName: string;
        objectId: string;
        mail: string;
    };
    lastChangedBy: {
        on: Date;
        displayName: string;
        objectId: string;
        mail: string;
    };
    entryType: BlogEntryType;
    status: BlogEntryStatus;
    body?: string; // only returned on detail API
    authors: 
        {
            name: string;
            gitHubAlias?: string;
            gitHubAvatar?: string;
            linkedin?: string;
        }[]
    ;
}

File: DeliveryIP_GitHub/dstoolkit/api/Microsoft.DataStrategy.Core/Models/BlogEntry.cs
using Newtonsoft.Json;

namespace Microsoft.DataStrategy.Core.Models
{
    public class BlogEntry : BlogEntryBase
    {
        [JsonProperty("body")]
        public string Body { get; set; }        
    }
    
    public class BlogEntryBase
    {
        // Year
        [JsonProperty("partitionKey")]
        public string PartitionKey;

        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("heroImage")]
        public Image HeroImage { get; set; }

        [JsonProperty("introduction")]
        public string Introduction { get; set; }

        [JsonProperty("createdBy")]
        public UserActionMetadata CreatedBy { get; set; }
        [JsonProperty("lastChangedBy")]
        public UserActionMetadata LastChangedBy { get; set; }

        [JsonProperty("entryType")]
        public BlogEntryType EntryType { get; set; }

        [JsonProperty("status")]
        public BlogEntryStatus Status { get; set; }
    }
  
}